### ISSUE TITLE
field-search: wrong number of search result on sidebar

### ISSUE THREAD

**Issue by kookma** [wikilabs/plugins#85]
TW: 5.1.24
Field-Search: 0.4.0-pre

Assume you have set up two configurations like Tags and Keywords
Then you make a search and click on the Tags tab you see zero results while if you click on list you see correct number of results!
If the drop down is closed and you do a search the result is displayed based on the active tab in search drop down!

Example

![image](https://user-images.githubusercontent.com/830394/112733857-f7621f00-8f5f-11eb-808e-621503fc56c1.png)

![img_23_msedge](https://user-images.githubusercontent.com/830394/112733891-385a3380-8f60-11eb-976c-f634d9686be7.png)

**Comment by pmario** [comment]
That's probably the same problem with tiddlywiki.com ... I'll have to check it. ...

**Comment by pmario** [comment]
It was easy to add the number in the dropdown. But the outer number makes some headache. There may be several tabs, which will show different results. So the truth would be "add all tabs without duplicates". ... BUT ... 

The actual number is only known if the tab is selected. ... So I personally would like to remove the "matches" text if an other tab as the default tab is selected. 

This means, that the core tiddler "$:/core/ui/SideBarSegments/search" needs to be changed. ... BUT I don't want to do that, since it's an ultra complicated one since V5.1.23 because of the keyboard handling. ... 

So the way to go imo will be to split this tiddler in the core, to make it easy to replace different elements by plugins.

**Comment by pmario** [comment]
I'll post a new version soon

**Comment by kookma** [comment]
Hi @pmario 
Yes I think the $:/core/ui/SideBarSegments/search need to be changes! If the core accept a simple field/index to be used by plugin developers to show the matches there! It could be enough to prevent overwritten the core $:/core/ui/SideBarSegments/search 

By the way, I think with current situation the drop down shall be opened to see the actual results! My headache is when the drop down is closed and I search something I look at the matches messages at the right and if it shows zero then I think there is no results at all!

**Comment by pmario** [comment]
>...My headache is when the drop down is closed and I search something I look at the matches messages at the right and if it shows zero then I think there is no results at all!

Yea, that's the problem. ... So my idea would be to have something like: (0/3/2/0) ... Where every number represents 1 tab. 
but
There is still a problem, if 1 tab has several fields. ... _and_ there is the problem, that those numbers would need to be recalculated all the times, even if the tab isn't shown. ... 

For some tiddlers, that may not be a problem, but if you have 10 thousands of tiddlers it will be. .. **So my initial thinking was, that I don't want to show the message if field-search tabs are selected.** ... That still needs help from the core

**Comment by kookma** [comment]
> For some tiddlers, that may not be a problem, but if you have 10 thousands of tiddlers it will be. .. **So my initial thinking was, that I don't want to show the message if field-search tabs are selected.** ... That still needs help from the core

Yes, that is the problem! What you think if only do not the zero matches e.g. if there is a result then we could notify somehow!
We see the correct results when we select a tab

**Comment by pmario** [comment]
Did push V0.4.1 have fun!
